Missing lua.hpp error during build

### Issue Summary
Latest commit results in missing lua.hpp error during build.

### Behavior/Symptoms
```
[ 82%] Building CXX object Projects/CoX/Servers/MapServer/CMakeFiles/mapserver_lib.dir/ScriptingEngine.cpp.o
In file included from /SEGS/Segs/Projects/CoX/Servers/MapServer/ScriptingEngine.cpp:10:0:
/SEGS/build/3rd_party/built/include/sol2/sol.hpp:1578:10: fatal error: lua.hpp: No such file or directory
 #include <lua.hpp>
          ^~~~~~~~~
compilation terminated.
gmake[2]: *** [Projects/CoX/Servers/MapServer/CMakeFiles/mapserver_lib.dir/build.make:615: Projects/CoX/Servers/MapServer/CMakeFiles/mapserver_lib.dir/ScriptingEngine.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1318: Projects/CoX/Servers/MapServer/CMakeFiles/mapserver_lib.dir/all] Error 2
gmake: *** [Makefile:84: all] Error 2
19:26:05: The process "/usr/bin/cmake" exited with code 2.
Error while building/deploying project SEGS (kit: Desktop)
When executing step "Make"
19:26:05: Elapsed time: 00:01.
```

### Steps to reproduce behavior/symptoms
Clone latest git (540e7bce4d11c5126360c8180af6b39b3aa818a6) and build.

### System Info
Qt 5.9.2 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 7.2.1 20170915 (Red Hat 7.2.1-2)) on "xcb" 
OS: Fedora 26 (Workstation Edition) [linux version 4.13.8-200.fc26.x86_64]

--- a/3rd_party/lua_sol2/include/sol.hpp
+++ b/3rd_party/lua_sol2/include/sol.hpp
@@ -1575,7 +1575,7 @@ namespace sol {
 #define SOL_EXCEPTIONS_SAFE_PROPAGATION
 #endif // Exceptions can be propagated safely using C++-compiled Lua
 #else
-#include <lua.hpp>
+#include <lua/lua.hpp>
 #endif // C++ Mangling for Lua
 
 #ifdef LUAJIT_VERSION

--- a/Projects/CoX/Common/NetStructures/Character.cpp
+++ b/Projects/CoX/Common/NetStructures/Character.cpp
@@ -122,9 +122,9 @@ void Character::GetCharBuildInfo(BitStream &src)
     m_level=0;
     src.GetString(m_class_name);
     src.GetString(m_origin_name);
-    PowerPool_Info primary,secondary;
-    primary.serializefrom(src);
-    secondary.serializefrom(src);
+    CharacterPower primary,secondary;
+    primary.power_id.serializefrom(src);
+    secondary.power_id.serializefrom(src);
 
     m_powers.push_back(primary); // primary_powerset power
     m_powers.push_back(secondary); // secondary_powerset power
@@ -132,40 +132,35 @@ void Character::GetCharBuildInfo(BitStream &src)
 }
 void Character::SendCharBuildInfo(BitStream &bs) const
 {
-    PowerPool_Info null_power;
-    null_power.id[0]=null_power.id[1]=null_power.id[2]=0;
+    PowerPool_Info null_power = {0,0,0};
     bs.StoreString(m_class_name); // class name
     bs.StoreString(m_origin_name); // origin name
     bs.StorePackedBits(5,0); // related to power level  ?
     PUTDEBUG("SendCharBuildInfo after plevel");
 
     {
-        //m_powers[0].serializeto(bs);
-        //m_powers[1].serializeto(bs);
         // TODO: this is character powers related, refactor it out of here.
         int count=0;
         bs.StorePackedBits(4,count); // count
         for(int i=0; i<count; i++)
         {
-            uint32_t num_powers=0;
+            uint32_t num_powers=m_powers.size();
             bs.StorePackedBits(5,0);
             bs.StorePackedBits(4,uint32_t(num_powers));
-            for(uint32_t idx=0; idx<num_powers; ++idx)
+            for(const CharacterPower &power : m_powers)
             {
-                uint32_t num_somethings=0;
-
                 //sendPower(bs,0,0,0);
-                m_powers[idx].serializeto(bs);
-                bs.StorePackedBits(5,0);
-                bs.StoreFloat(1.0);
-                bs.StorePackedBits(4,num_somethings);
+                power.power_id.serializeto(bs);
+                bs.StorePackedBits(5,power.bought_at_level);
+                bs.StoreFloat(power.range);
+                bs.StorePackedBits(4,power.boosts.size());
 
-                for(size_t idx2=0; idx2<num_somethings; ++idx2)
+                for(const CharacterPowerBoost &boost : power.boosts)
                 {
                     //sendPower(bs,0,0,0);
-                    m_powers[idx2].serializeto(bs);
-                    bs.StorePackedBits(5,0);
-                    bs.StorePackedBits(2,0);
+                    boost.boost_id.serializeto(bs);
+                    bs.StorePackedBits(5,boost.level);
+                    bs.StorePackedBits(2,boost.num_combines);
                 }
             }
         }
@@ -273,17 +268,22 @@ void Character::serialize_costumes(BitStream &bs, ColorAndPartPacker *packer , b
 }
 void Character::DumpPowerPoolInfo( const PowerPool_Info &pool_info )
 {
-    for(int i=0; i<3; i++)
+    for (int i = 0; i < 3; i++)
     {
-        qDebug().nospace().noquote() << "    Pool_id["<<i<<"]: 0x%"<<QString::number(pool_info.id[i],16);
+        qDebug().nospace().noquote() << "    "
+                                     << QString("Pool_id[%1]: %2 %3 %4")
+                                            .arg(i)
+                                            .arg(pool_info.category_idx)
+                                            .arg(pool_info.powerset_entry_idx)
+                                            .arg(pool_info.power_idx);
     }
 }
 void Character::DumpBuildInfo()
 {
     qDebug() << "    class: "<<m_class_name;
     qDebug() << "    origin: "<<m_origin_name;
-    DumpPowerPoolInfo(m_powers[0]);
-    DumpPowerPoolInfo(m_powers[1]);
+    DumpPowerPoolInfo(m_powers[0].power_id);
+    DumpPowerPoolInfo(m_powers[1].power_id);
 }
 
 void Character::dump()

--- a/Projects/CoX/Common/NetStructures/Character.h
+++ b/Projects/CoX/Common/NetStructures/Character.h
@@ -93,12 +93,26 @@ public:
 
 };
 struct Costume;
+struct CharacterPowerBoost
+{
+    PowerPool_Info boost_id;
+    int            level        = 0;
+    int            num_combines = 0;
+};
+struct CharacterPower
+{
+    PowerPool_Info                   power_id;
+    int                              bought_at_level = 0;
+    float                            range           = 1.0f;
+    std::vector<CharacterPowerBoost> boosts;
+};
 
 class Character
 {
-friend  class CharacterDatabase;
-typedef std::vector<PowerPool_Info> vPowerPool;
-typedef std::vector<Costume *> vCostume;
+        friend  class CharacterDatabase;
+
+        using vPowerPool = std::vector<CharacterPower>;
+        using vCostume = std::vector<Costume *>;
 
         vPowerPool      m_powers;
         PowerTrayGroup  m_trays;

--- a/Projects/CoX/Common/NetStructures/Powers.cpp
+++ b/Projects/CoX/Common/NetStructures/Powers.cpp
@@ -162,14 +162,14 @@ void PowerTray::Dump()
 
 void PowerPool_Info::serializefrom(BitStream &src)
 {
-    id[0] = src.GetPackedBits(3);
-    id[1] = src.GetPackedBits(3);
-    id[2] = src.GetPackedBits(3);
+    category_idx = src.GetPackedBits(3);
+    powerset_entry_idx = src.GetPackedBits(3);
+    power_idx = src.GetPackedBits(3);
 }
 
 void PowerPool_Info::serializeto(BitStream &src) const
 {
-    src.StorePackedBits(3,id[0]);
-    src.StorePackedBits(3,id[1]);
-    src.StorePackedBits(3,id[2]);
+    src.StorePackedBits(3,category_idx);
+    src.StorePackedBits(3,powerset_entry_idx);
+    src.StorePackedBits(3,power_idx);
 }

--- a/Projects/CoX/Common/NetStructures/Powers.h
+++ b/Projects/CoX/Common/NetStructures/Powers.h
@@ -25,7 +25,9 @@ enum class TrayItemType : uint32_t
 class PowerPool_Info
 {
 public:
-    int id[3];
+    int category_idx;
+    int powerset_entry_idx;
+    int power_idx;
     void serializefrom( BitStream &src );
     void serializeto( BitStream &src ) const;
 };

--- a/Projects/CoX/Servers/MapServer/ScriptingEngine.h
+++ b/Projects/CoX/Servers/MapServer/ScriptingEngine.h
@@ -1,7 +1,7 @@
 #ifndef SCRIPTINGENGINE_H
 #define SCRIPTINGENGINE_H
 #include <memory>
-
+#include <string>
 class QString;
 class MapClient;
 

